src/vs/workbench/contrib/snippets/browser/commands/configureSnippets.ts

```diff
@@ -66,13 +73,19 @@ async function computePicks(snippetService: ISnippetsService, userDataProfileSer
 				}
 			}
 
+			let desc = names.size === 0
+				? nls.localize('global.scope', "(global)")
+				: nls.localize('global.1', "({0})", [...names].join(', '));
+			if (added.has(basename(file.location)) && source) {
+				desc = source + ' ' + desc;
+			}
+
 			existing.push({
 				label: basename(file.location),
 				filepath: file.location,
-				description: names.size === 0
-					? nls.localize('global.scope', "(global)")
-					: nls.localize('global.1', "({0})", [...names].join(', '))
+				description: desc
 			});
+			added.add(basename(file.location));
```

Comment:
This will "miss" the first item, e.g only the second, third etc will get the more detailed label but not the second. You can make `added` be a map of basename to last item and always update the last seen item and the current one.

Reply:
I'm not sure if I want to do a check on 'snippet.source' all the time, and if it definitely has value, I guess the code will remove some